azurerm/resource_arm_loadbalancer_outbound_rule.go

```diff
@@ -0,0 +1,346 @@
+		if *OutboundRule.Name == name {
+			outboundRuleId = *OutboundRule.ID
+		}
+	}
+
+	if outboundRuleId == "" {
+		return fmt.Errorf("Cannot find created Load Balancer Outbound Rule ID %q", outboundRuleId)
+	}
+
+	d.SetId(outboundRuleId)
+
+	return resourceArmLoadBalancerOutboundRuleRead(d, meta)
+}
+
+func resourceArmLoadBalancerOutboundRuleRead(d *schema.ResourceData, meta interface{}) error {
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	name := id.Path["outboundRules"]
+
+	loadBalancer, exists, err := retrieveLoadBalancerById(d.Get("loadbalancer_id").(string), meta)
+	if err != nil {
+		return fmt.Errorf("Error Getting Load Balancer By ID: %+v", err)
+	}
+	if !exists {
+		d.SetId("")
+		log.Printf("[INFO] Load Balancer %q not found. Removing from state", name)
+		return nil
+	}
+
+	config, _, exists := findLoadBalancerOutboundRuleByName(loadBalancer, name)
+	if !exists {
+		d.SetId("")
+		log.Printf("[INFO] Load Balancer Outbound Rule %q not found. Removing from state", name)
+		return nil
+	}
+
+	d.Set("name", config.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+
+	if properties := config.OutboundRulePropertiesFormat; properties != nil {
+		d.Set("protocol", properties.Protocol)
+		d.Set("backend_address_pool_id", properties.BackendAddressPool.ID)
+
+		frontendIpConfigurations := make([]interface{}, 0)
+		for _, feConfig := range *properties.FrontendIPConfigurations {
+			if feConfig.ID == nil {
+				continue
+			}
+
+			feConfigId, err := parseAzureResourceID(*feConfig.ID)
+			if err != nil {
+				return nil
+			}
+
+			name := feConfigId.Path["frontendIPConfigurations"]
+			frontendConfiguration := map[string]interface{}{
+				"id":   *feConfig.ID,
+				"name": name,
+			}
+			frontendIpConfigurations = append(frontendIpConfigurations, frontendConfiguration)
+		}
+		d.Set("frontend_ip_configuration", frontendIpConfigurations)
+
+		if properties.EnableTCPReset != nil {
+			d.Set("enable_tcp_reset", properties.EnableTCPReset)
+		}
+
+		if properties.IdleTimeoutInMinutes != nil {
+			d.Set("idle_timeout_in_minutes", properties.IdleTimeoutInMinutes)
+		}
+
+		if properties.AllocatedOutboundPorts != nil {
+			d.Set("allocated_outbound_ports", properties.AllocatedOutboundPorts)
+		}
+
+	}
+
+	return nil
+}
+
+func resourceArmLoadBalancerOutboundRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).loadBalancerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	loadBalancerID := d.Get("loadbalancer_id").(string)
+	armMutexKV.Lock(loadBalancerID)
+	defer armMutexKV.Unlock(loadBalancerID)
+
+	loadBalancer, exists, err := retrieveLoadBalancerById(loadBalancerID, meta)
+	if err != nil {
+		return fmt.Errorf("Error Getting Load Balancer By ID: %+v", err)
+	}
+	if !exists {
+		d.SetId("")
+		return nil
+	}
+
+	_, index, exists := findLoadBalancerOutboundRuleByName(loadBalancer, d.Get("name").(string))
+	if !exists {
```

Comment:
Minor but these two lines could be combined:
```
	if _, index, exists := findLoadBalancerOutboundRuleByName(loadBalancer, d.Get("name").(string)); !exists {
```

Reply:
since I'm using index later on, I think that I can't combine those two lines.